### PR TITLE
﻿Adding office aliases, and handling formats

### LINKED ISSUE #503: Update endpoint: offices
Add deprecated header to original formats:

- CSV
- TAB
- JSON/XML from pl/sql

### PR DESCRIPTION
**Pull request by RyanM-RMA**
Updated parseHeaderAndQueryParm method to include the class Updated unit tests for parseQueryParam and updated it to include the class for aliases
Fixes #503

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/api/OfficeController.java [hunk 1]
```diff
@@ -121,17 +122,23 @@ public void getAll(Context ctx) {
...
-                            + "format of the response. Valid value for the format field for this "
-                            + "URI are:\r\n1. tab\r\n2. csv\r\n 3. xml\r\n4. json (default)"
+                    description = "(CWMS-Data-Format-Deprecated: 2024-11-01 in favor of Accept header)"
+                            + " Specifies the encoding format of the response. Valid value for the format "
+                            + "field for this URI are:\r\n"
+                                + "1. `tab`\r\n"
+                                + "2. `csv`\r\n"
+                                + "3. `xml`\r\n"
+                                + "4. `json` (default)"
             ),
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/api/OfficeController.java]: Remove the numbered list format here and make it a bullet list like in getAll.   I thought I replaced that everywhere but I guess I missed this one.

File: cwms-data-api/src/main/java/cwms/cda/data/dto/Office.java [hunk 2]
```diff
@@ -19,10 +19,12 @@
...
 @JsonRootName("office")
 @JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
-@FormattableWith(contentType = Formats.XML, formatter = XMLv1.class)
-@FormattableWith(contentType = Formats.XMLV2, formatter = XMLv2.class)
-@FormattableWith(contentType = Formats.JSON, formatter = JsonV1.class)
-@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@FormattableWith(contentType = Formats.XML, formatter = XMLv2.class)
+@FormattableWith(contentType = Formats.XMLV1, formatter = XMLv1.class)
+@FormattableWith(contentType = Formats.XMLV2, formatter = XMLv2.class, aliases = {Formats.XML})
+@FormattableWith(contentType = Formats.JSON, formatter = JsonV2.class)
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/Office.java]: If JSON is an alias for JsonV2 do you need this line?

Comment (RyanM-RMA) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/Office.java]: Ah, good point, don't need that, or XML

File: cwms-data-api/src/main/java/cwms/cda/api/OfficeController.java (hunk 1 above)
Comment (RyanM-RMA) [reply cwms-data-api/src/main/java/cwms/cda/api/OfficeController.java]: Updated to bullet points, I'll push once you're done reviewing

**Review by rma-rripken — DISMISSED** [review]

**Review by rma-rripken — APPROVED** [review]

**Pull request merged**